apps/openmw/mwworld/worldimp.cpp

```diff
@@ -154,23 +154,16 @@ namespace MWWorld
       mLevitationEnabled(true), mGoToJail(false), mDaysInPrison(0),
       mPlayerTraveling(false), mPlayerInJail(false), mSpellPreloadTimer(0.f)
     {
-        mEsm.resize(contentFiles.size() + groundcoverFiles.size());
+        mEsm.resize(contentFiles.size());
         Loading::Listener* listener = MWBase::Environment::get().getWindowManager()->getLoadingScreen();
         listener->loadingOn();
-
-        GameContentLoader gameContentLoader(*listener);
-        EsmLoader esmLoader(mStore, mEsm, encoder, *listener);
-
-        gameContentLoader.addLoader(".esm", &esmLoader);
-        gameContentLoader.addLoader(".esp", &esmLoader);
-        gameContentLoader.addLoader(".omwgame", &esmLoader);
-        gameContentLoader.addLoader(".omwaddon", &esmLoader);
-        gameContentLoader.addLoader(".project", &esmLoader);
-
-        OMWScriptsLoader omwScriptsLoader(*listener, mStore);
-        gameContentLoader.addLoader(".omwscripts", &omwScriptsLoader);
-
-        loadContentFiles(fileCollections, contentFiles, groundcoverFiles, gameContentLoader);
+        
+        loadContentFiles(fileCollections, contentFiles, mStore, mEsm, encoder, listener);
+        if (!groundcoverFiles.empty())
+        {
+            std::vector<ESM::ESMReader> tempReaders (groundcoverFiles.size());
+            loadContentFiles(fileCollections, groundcoverFiles, mGroundcoverStore, tempReaders, encoder, listener);
```

Comment:
In my testing the game spends some seconds here to load groundcover ESPs.

Reply:
I have received feedback indicating that optimisations specific to legacy groundcover are not univocally appreciated because they might discourage the inception of a more suitable replacement.